session.go

```diff
@@ -123,8 +123,11 @@ func (h *StmtHistory) Count() int {
 type session struct {
 	// processInfo is used by ShowProcess(), and should be modified atomically.
 	processInfo atomic.Value
-	txn         kv.Transaction // current transaction
-	txnFuture   *txnFuture
+
+	// TODO: Make txnFuture into StmtTxn to optimize the txn representation, avoid
+	// the ugly code like if s.Transaction == nil || s.Transaction.Valid()
+	StmtTxn
```

Comment:
Why make `StmtTxn` embedded?
Logically, a session is not a StmtTxn.

Reply:
Because this makes the minimal changes, just 

```
t.txn != nil && t.txn.Valid()
```

to 

```
t.Transaction != nil && t.Transaction.Valid()
```

I don't want introduce risky modification in this PR.
I'll refine the code and remove the TODO above in the following PRs.

Reply:
We can add a method like `baseTxn()` to return the underlying transaction, and we can easily see it's different than `Txn()`.

Reply:
We can also do it like this:

```
type session {
    txn StmtTxn
}

type (txn *StmtTxn) Valid() {
}
```

The source of the problem is that we need verify null point and also txn state is valid.
If the StmtTxn would always not null, and provide Valid() method, the code can be much more elegant.